am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,128 @@
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class,
+            dao -> dao.addRole(roleName, roleType, securityClassification, accessManagementType));
+    }
+
+    public void addResourceDefinition(@NonNull String serviceName,
+                                      @NonNull String resourceType,
+                                      @NonNull String resourceName) {
+        if (serviceName.isEmpty()) {
+            throw new IllegalArgumentException("Service name cannot be empty");
+        }
+
+        if (resourceType.isEmpty() || resourceName.isEmpty()) {
+            throw new IllegalArgumentException("Resource cannot contain empty values");
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class, dao ->
+            dao.addResourceDefinition(serviceName, resourceType, resourceName));
+
+    }
+
+    public void grantDefaultPermission(DefaultPermissionGrant defaultPermissionGrant) {
+
+        jdbi.useTransaction((handle) -> {
+            DefaultRoleSetupRepository dao = handle.attach(DefaultRoleSetupRepository.class);
+            try {
+                defaultPermissionGrant.getAttributePermissions().forEach((attribute, permissionAndClassification) -> {
+                    dao.createResourceAttribute(
+                        defaultPermissionGrant.getServiceName(),
+                        defaultPermissionGrant.getResourceType(),
+                        defaultPermissionGrant.getResourceName(),
+                        attribute.toString(),
+                        permissionAndClassification.getValue());
+
+                    dao.grantDefaultPermission(
+                        DefaultPermission.builder()
+                            .serviceName(defaultPermissionGrant.getServiceName())
+                            .resourceType(defaultPermissionGrant.getResourceType())
+                            .resourceName(defaultPermissionGrant.getResourceName())
+                            .attribute(attribute.toString())
+                            .roleName(defaultPermissionGrant.getRoleName())
+                            .permissions(Permissions.sumOf(permissionAndClassification.getKey()))
+                            .build());
+                });
+
+            } catch (Exception e) {
+                throw new ErrorAddingEntriesToDatabaseException(e);
+            }
+        });
+    }
+
+    public void truncateAllDefaultPermissionsForService(String serviceName, String resourceType) {
+        jdbi.useTransaction((handle) -> {
+            try {
+                DefaultRoleSetupRepository dao = handle.attach(DefaultRoleSetupRepository.class);
+                dao.deleteDefaultPermissionsForRoles(serviceName, resourceType);
+                dao.deleteResourceAttributes(serviceName, resourceType);
+            } catch (Exception e) {
+                //TODO: new exception for removing entries.
+                throw new ErrorAddingEntriesToDatabaseException(e);
```

Comment:
Please make more generic exception. You can even use name that does not mention database in case we wanted to use something else for storage at some point.

Reply:
@geek-soft can I just double check that this would be a more generic exception for removing things from the db? Any suggestions? I was thinking ErrorRemovingEntriesFromDb where the message would read something like "error, transaction has been rolled back".